src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobMetadataStore.cs

```diff
@@ -130,14 +129,12 @@ public Task<DicomDataset> GetInstanceMetadataAsync(VersionedInstanceIdentifier v
 
         return ExecuteAsync(async t =>
         {
-            await using (Stream stream = _recyclableMemoryStreamManager.GetStream(GetInstanceMetadataStreamTagName))
-            {
-                await blobClient.DownloadToAsync(stream, cancellationToken);
-
-                stream.Seek(0, SeekOrigin.Begin);
+            BlobDownloadResult result = await blobClient.DownloadContentAsync(t);
 
-                return await JsonSerializer.DeserializeAsync<DicomDataset>(stream, _jsonSerializerOptions, t);
-            }
+            // DICOM metadata file was saved using UTF-8 encoding with BOM, When using JsonSerializer, by design UTF-8 BOM should be ignored
```

Comment:
I would just say that the DICOM metadata file includes the UTF-8 BOM and that there is a bug with the `BinaryData.ToObjectFromJson` method as seen in this issue: https://github.com/dotnet/runtime/issues/71447. The other linked issue doesn't seem like a real issue lol

Reply:
Updated comment